tests/images/run.sh

```diff
@@ -34,3 +34,10 @@ done
 for png_filename in tests/images/imagenet/*.png; do
   ./bin/image-classifier "$png_filename" -image_mode=0to256 -m=bvlc_alexnet/model.onnx -model_input_name=data_0 "$@"
 done
+#Inference test for TF ONNX models
+for png_filename in tests/images/imagenet/*.png; do
+  ./bin/image-classifier "$png_filename" -image_mode=0to1 -m=googlenet_v1_slim/googlenet_v1_slim.onnx -model_input_name=input:0 -image_layout=NHWC "$@"
+done
+for png_filename in tests/images/imagenet_299/*.png; do
```

Comment:
googlenet_v4 needs input images of size 299x299x3 and not 224x224x3 which are the sizes of the images in the imagenet folder. Hence the new directory called imagenet_299. Kindly advise if you want these structured in a different way.

Reply:
Also, this is a screenshot of the inference results. The TF models were trained on 1001 classes. So I see the results are off by for the last 6 test cases that were added by me as compared to an inference made on tensorflow directly. Is there any way I can adjust this?

Model: inception_v1/model.onnx
 File: tests/images/imagenet/cat_285.png Result:281
Model: inception_v1/model.onnx
 File: tests/images/imagenet/dog_207.png Result:207
Model: inception_v1/model.onnx
 File: tests/images/imagenet/zebra_340.png Result:340
Model: bvlc_alexnet/model.onnx
 File: tests/images/imagenet/cat_285.png Result:281
Model: bvlc_alexnet/model.onnx
 File: tests/images/imagenet/dog_207.png Result:207
Model: bvlc_alexnet/model.onnx
 File: tests/images/imagenet/zebra_340.png Result:340
Model: googlenet_v1_slim/googlenet_v1_slim.onnx
 File: tests/images/imagenet/cat_285.png Result:282
Model: googlenet_v1_slim/googlenet_v1_slim.onnx
 File: tests/images/imagenet/dog_207.png Result:223
Model: googlenet_v1_slim/googlenet_v1_slim.onnx
 File: tests/images/imagenet/zebra_340.png Result:341
Model: googlenet_v4_slim/googlenet_v4_slim.onnx
 File: tests/images/imagenet_299/cat_285_299.png Result:282
Model: googlenet_v4_slim/googlenet_v4_slim.onnx
 File: tests/images/imagenet_299/dog_207_299.png Result:208
Model: googlenet_v4_slim/googlenet_v4_slim.onnx
 File: tests/images/imagenet_299/zebra_340_299.png Result:341

Reply:
The inference result is off by 1 as caffe onnx has 1000 labels but the TF pretrained model has 1001 labels

Reply:
You can hand-modify TF ONNX models. Add a SliceNode at the very end, to cut out the first element, bring the tensor to size of 1000.

But why do they have 1001 classes? Do you see the same results when running through TF? Or is this 1 extra class somehow added by converter?

Reply:
It turns out that the inception function makes a prediction for a background class in addition to the 1000 classes in Imagenet. I will see how I can tweak the model, thanks

Reply:
@artemrakhov Running through TF would give the same results (this index shift is not introduced by the ONNX converter).  Some pre-trained models consider a label indexes range [0:999] while some others consider range [1:1000].   The index range for labels is a training choice and is not specific to TF.

Reply:
I see that tensorflow slim has addressed this between the models by adding a label_offset. Is there an option to add a CLI in the GLOW inferencing to handle this? I agree tweaking intermediate ONNX models can be a very custom approach per model and introduces another step in between.

The ResNet and VGG Models have 1000 classes but the ImageNet dataset has 1001
The ImageNet dataset provided has an empty background class which can be used to fine-tune the model to other tasks. If you try training or fine-tuning the VGG or ResNet models using the ImageNet dataset, you might encounter the following error:

InvalidArgumentError: Assign requires shapes of both tensors to match. lhs shape= [1001] rhs shape= [1000]
This is due to the fact that the VGG and ResNet V1 final layers have only 1000 outputs rather than 1001.

To fix this issue, you can set the --labels_offset=1 flag. This results in the ImageNet labels being shifted down by one:

Reply:
@dmoham1476 I think that adding a `--labels_offset` option to `image-classifier` would be a good solution. It doesn't require any change deep in glow or in the input model, and it's up to the user to choose how he wants the results displayed by `image-classifier`

Reply:
I support adding `--labels_offset` option (flag) to `image-classifier`. Defaults to 0, but will be 1 for the models coming from TF.

Reply:
In Image Classifier, which of these functions retrieve the label from the inferred value or is the output already a label index? In cifar10 and mnist, I see a labelH that does the guess vs actual comparison. I think its better to modify the actual with the label offset than manipulate the guess itself. 
Variable *SMVar = LD->getSingleOutput();
Tensor &res = SMVar->getPayload();
Tensor slice = H.extractSlice(i);
SH.minMaxArg().second //this is the final index in the labels file

include/glow/Importer/ProtobufLoader.h:  llvm::StringMap<Variable *> outputVarsByName_; I see a StringMap here.

In other

Reply:
If I understand the question correctly, `SH.minMaxArg().second;` gets the label.  If it's really as simple as the guess being shifted by one, I'd recommend simply offsetting the guess before output.

Reply:
Thanks for this simple suggestion